### DISCUSSION TITLE
Why doesn't self.log.info("Hello World") or print("Hello World") work in the __init__ method of my Custom Operator?

### DISCUSSION THREAD

**Discussion by sonjeongwons** [apache/airflow#44649] [Q&A]
After inheriting BaseOperator, we are creating a custom operator.

Why doesn't self.log.info("Hello World") or print("Hello World") work in the __init__ method of my Custom Operator?

Those logging codes work in the pre_execute(), execute(), and post_execute() methods. However, they don't work in the __init__() method. Why is that?

**Comment by potiuk** [comment]
Because operator is created when DAG structure is created - and the others are run when the task is executed. This is basic assumpion - see also https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code. The `__init__` of your operator - unlike the other methods you mentioned - are "top level code" for your DAG and they are executed in parse time not "Task execution" time.
